### ISSUE TITLE
`model_parameters` fails with `truncreg` models

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/parameters#188]
This was working. Not sure what changed.

``` r
# setup
set.seed(123)
library(truncreg)
#> Loading required package: maxLik
#> Loading required package: miscTools
#> 
#> Please cite the 'maxLik' package as:
#> Henningsen, Arne and Toomet, Ott (2011). maxLik: A package for maximum likelihood estimation in R. Computational Statistics 26(3), 443-458. DOI 10.1007/s00180-010-0217-1.
#> 
#> If you have questions, suggestions, or comments regarding the 'maxLik' package, please use a forum or 'tracker' at maxLik's R-Forge site:
#> https://r-forge.r-project.org/projects/maxlik/
library(survival)

# data
data("tobin", package = "survival")

# model
cragg_trunc <- truncreg(
  formula = durable ~ age + quant,
  data = tobin,
  subset = durable > 0
)

# exctract parameters
parameters::model_parameters(cragg_trunc)
#> Error in fix.by(by.y, y): 'by' must specify a uniquely valid column
```

<sup>Created on 2020-01-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

<details style="margin-bottom:10px;">

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.2 (2019-12-12)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       Europe/Berlin               
#>  date     2020-01-01                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version    date       lib source                           
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.6.0)                   
#>  backports     1.1.5      2019-10-02 [1] CRAN (R 3.6.1)                   
#>  bayestestR    0.4.0      2019-10-20 [1] CRAN (R 3.6.1)                   
#>  cli           2.0.0.9000 2019-12-23 [1] Github (r-lib/cli@0293ae7)       
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.1)                   
#>  digest        0.6.23     2019-11-23 [1] CRAN (R 3.6.1)                   
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 3.6.0)                   
#>  fansi         0.4.0      2018-11-05 [1] Github (brodieG/fansi@ab11e9c)   
#>  fs            1.3.1      2019-05-06 [1] CRAN (R 3.6.0)                   
#>  glue          1.3.1      2019-03-12 [1] CRAN (R 3.6.0)                   
#>  highr         0.8        2019-03-20 [1] CRAN (R 3.6.0)                   
#>  htmltools     0.4.0      2019-10-04 [1] CRAN (R 3.6.1)                   
#>  insight       0.7.1.1    2020-01-01 [1] local                            
#>  knitr         1.26       2019-11-12 [1] CRAN (R 3.6.1)                   
#>  lattice       0.20-38    2018-11-04 [2] CRAN (R 3.6.2)                   
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.1)                   
#>  Matrix        1.2-18     2019-11-27 [1] CRAN (R 3.6.1)                   
#>  maxLik      * 1.3-6      2019-05-19 [1] CRAN (R 3.6.0)                   
#>  miscTools   * 0.6-26     2019-12-08 [1] CRAN (R 3.6.1)                   
#>  parameters    0.3.0.1    2020-01-01 [1] local                            
#>  pillar        1.4.3      2019-12-20 [1] CRAN (R 3.6.2)                   
#>  pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 3.6.1)                   
#>  purrr         0.3.3      2019-10-18 [1] CRAN (R 3.6.1)                   
#>  Rcpp          1.0.3      2019-11-08 [1] CRAN (R 3.6.1)                   
#>  reprex        0.3.0.9001 2019-12-30 [1] Github (tidyverse/reprex@27aa69a)
#>  rlang         0.4.2      2019-11-23 [1] CRAN (R 3.6.2)                   
#>  rmarkdown     2.0        2019-12-12 [1] CRAN (R 3.6.1)                   
#>  rstudioapi    0.10       2019-03-19 [1] CRAN (R 3.6.0)                   
#>  sandwich      2.5-1      2019-04-06 [1] CRAN (R 3.6.0)                   
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)                   
#>  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.6.0)                   
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.6.0)                   
#>  styler        1.2.0.9000 2019-12-20 [1] Github (r-lib/styler@9c6346c)    
#>  survival    * 3.1-8      2019-12-03 [1] CRAN (R 3.6.1)                   
#>  tibble        2.1.3      2019-06-06 [1] CRAN (R 3.6.2)                   
#>  truncreg    * 0.2-5      2018-08-18 [1] CRAN (R 3.6.2)                   
#>  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.1)                   
#>  xfun          0.11       2019-11-12 [1] CRAN (R 3.6.1)                   
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                   
#>  zoo           1.8-6      2019-05-28 [1] CRAN (R 3.6.0)                   
#> 
#> [1] C:/Users/inp099/Documents/R/win-library/3.6
#> [2] C:/Program Files/R/R-3.6.2/library
```

</details>

Trace-
```r
Error in fix.by(by.y, y) : 'by' must specify a uniquely valid column 
8.
stop(ngettext(sum(bad), "'by' must specify a uniquely valid column", 
    "'by' must specify uniquely valid columns"), domain = NA) 
7.
fix.by(by.y, y) 
6.
merge.data.frame(parameters, p_value(model, effects = effects, 
    component = component, method = robust), by = merge_by) 
5.
merge(parameters, p_value(model, effects = effects, component = component, 
    method = robust), by = merge_by) at extract_parameters.R#64
4.
.extract_parameters_generic(model, ci = ci, component = "conditional", 
    merge_by = merge_by, standardize = standardize, effects = effects, 
    robust = robust, ...) at model_parameters_default.R#70
3.
.model_parameters_generic(model = model, ci = ci, bootstrap = bootstrap, 
    iterations = iterations, merge_by = "Parameter", standardize = standardize, 
    exponentiate = exponentiate, effects = "fixed", robust = robust, 
    ...) at model_parameters_default.R#40
2.
model_parameters.default(cragg_trunc) at model_parameters.R#38
1.
parameters::model_parameters(cragg_trunc) 
```

**Comment by IndrajeetPatil** [comment]
We need to add tests for these types of model objects.

**Comment by strengejacke** [comment]
I removed some namespace-methods for classed that use the default method (https://github.com/easystats/parameters/commit/8c7b1a8f13939b0017f6ea364d600987a53f4690). This was because checks complained about to many S3-methods. However, some model objects have two classes and do not call the default-method if not explicitely assigned (like truncreg, which then called `.maxLik` instead of `.default`.) I reverted the change and set the requirement to R > 3.2, which should fix the check issue.
